### PR TITLE
Move decimal rounding to the EE (only default behavior).

### PR DESCRIPTION
**Pull request by BillWhite**
When we create a decimal number from a string in the EE,
and the resulting decimal number would have a scale which
is too large, we round the decimal number to 12 decimal digits.
If the resulting rounded number is still too large we throw
and exception.

There are java unit tests and C++ unit tests.

### CODE REVIEW

File: src/ee/common/NValue.cpp [hunk 1]
```diff
@@ -215,6 +214,7 @@ std::string NValue::createStringFromDecimal() const {
  *   This function does not handle scientific notation string, Java planner should convert that to plan string first.
  */
 void NValue::createDecimalFromString(const std::string &txt) {
+    std::cout << "createDecimalFromString(" << txt << ");\n";
```
Comment (jxchinaren) [inline src/ee/common/NValue.cpp]: Do you need this ?

File: src/ee/common/NValue.cpp [hunk 2]
```diff
@@ -259,27 +259,54 @@ void NValue::createDecimalFromString(const std::string &txt) {
...
     while (fractionalString.size() > 0 && fractionalString[fractionalString.size() - 1] == '0')
         fractionalString.erase(fractionalString.size() - 1, 1);
     // check if too many decimal places
-    if (fractionalString.size() > 12) {
-        throw SQLException(SQLException::volt_decimal_serialization_error,
-                           "Maximum scale exceeded. Maximum of 12 digits to the right of the decimal point");
-    }
-    while(fractionalString.size() < NValue::kMaxDecScale) {
-        fractionalString.push_back('0');
+    if (fractionalString.size() > kMaxDecScale) {
```
Comment (jxchinaren) [inline src/ee/common/NValue.cpp]: // Can you add a comment for using the default rounding here?

File: tests/frontend/org/voltdb/regressionsuites/TestDecimalRoundingSuite.java [hunk 3]
```diff
@@ -467,7 +560,7 @@ private static void addConfig(int idx, MultiConfigSuiteBuilder builder, VoltProj
         final String literalSchema =
                 "CREATE TABLE P1 ( id integer );" +
                 "CREATE TABLE DECIMALTABLE ( " +
-                "dec decimal" +
+                    "dec decimal"     +
```
Comment (jxchinaren) [inline tests/frontend/org/voltdb/regressionsuites/TestDecimalRoundingSuite.java]: Do we have any default decimal string value rounding tests? I guess the default value here also use interpreted from string, but probably at Java level? I am not sure.

End of review round.

Comment (BillWhite) [inline tests/frontend/org/voltdb/regressionsuites/TestDecimalRoundingSuite.java]: Actually, we have tests for all the rounding modes, and for rounding
enabled and disabled in the junit tests.  But this exercise is just to
implement the default, which is rounding enabled and HALF_UP.

File: src/ee/common/NValue.cpp (hunk 2 above)
Comment (BillWhite) [inline src/ee/common/NValue.cpp]: Done.

**Comment by BillWhite** [comment]
Close to delete this branch and push again with changes.

**Pull request closed without merging**